lib/prometheus_exporter/client.rb

```diff
@@ -107,7 +107,7 @@ def find_registered_metric(name, type: nil, help: nil)
     end
 
     def send_json(obj)
-      payload = @custom_labels.nil? ? obj : obj.merge(custom_labels: @custom_labels)
+      payload = @custom_labels.nil? ? obj : obj.merge(custom_labels: @custom_labels.merge(obj[:custom_labels] || {}))
```

Comment:
minor but this is perf sensitive. No need to merge here if obj[:custom_labels] is nil. 

```
payload = 
  if @custom_labels
     if obj[:custom_labels]
         ...
     else
        ...
     end 
  else
    obj
  end
  
```